openapi/api-pecserver.yaml

```diff
@@ -0,0 +1,196 @@
+          description: Success
+          schema:
+            $ref: '#/definitions/LegalMessage'
+        '400':
+          description: Bad Request
+        '401':
+          description: Unauthorized
+        '403':
+          description: Forbidden
+        '429':
+          description: Too Many Requests
+        '500':
+          description: Internal Server Error
+      security:
+        - bearerAuth: []
+      summary: Retrieve a legal message
+  '/messages/{id}/attachments/{attachment_id}':
+    get:
+      operationId: getMessageAttachment
+      produces:
+        - application/octet-stream
+      parameters:
+        - in: path
+          name: id
+          required: true
+          type: string
+        - in: path
+          name: attachment_id
+          required: true
+          type: string
+      responses:
+        '200':
+          description: Success
+          schema:
+            format: binary
+            type: string
+        '400':
+          description: Bad Request
+        '401':
+          description: Unauthorized
+        '403':
+          description: Forbidden
+        '429':
+          description: Too Many Requests
+        '500':
+          description: Internal Server Error
+      security:
+        - bearerAuth: []
+      summary: Retrieve an attachment of a legal message
+  '/messages/{id}/{format}':
+    get:
+      operationId: getMessagePart
+      produces:
+        - text/plain
+      parameters:
+        - in: path
+          name: id
+          required: true
+          type: string
+        - enum:
+            - eml
+            - smime
+            - daticert
+          in: path
+          name: format
+          type: string
+      responses:
+        '200':
+          description: Success
+          schema:
+            type: string
+        '400':
+          description: Bad Request
+        '401':
+          description: Unauthorized
+        '403':
+          description: Forbidden
+        '429':
+          description: Too Many Requests
+        '500':
+          description: Internal Server Error
+      security:
+        - bearerAuth: []
+      summary: Retrieve the original part of a legal message
+definitions:
+  Attachment:
+    properties:
+      content_type:
+        type: string
+      id:
+        type: string
+      name:
+        type: string
+      url:
+        type: string
+    type: object
+  CertData:
+    properties:
+      envelope_id:
+        minLength: 1
+        type: string
+      msg_id:
+        minLength: 1
+        type: string
+      receipt_type:
+        type: string
+      sender_provider:
+        minLength: 1
+        type: string
+      timestamp:
+        format: date-time
+        type: string
+    required:
+      - sender_provider
+      - timestamp
+      - envelope_id
+      - msg_id
+      - receipt_type
+    type: object
+  CertDataHeader:
+    properties:
+      object:
+        minLength: 1
+        type: string
+      recipients:
```

Comment:
Is it ok that it is recipient`s` and it is a string?
if it is intended as multiple mail address, I would have expected an array of strings (or `EmailAddress`)

Reply:
You are right, an array is more self-defined.